src/renderer/components/input/input.tsx

```diff
@@ -262,6 +272,10 @@ export class Input extends React.Component<InputProps, State> {
         this.props.onSubmit?.(this.getValue(), evt);
         this.setDirtyOnChange.cancel();
         this.setState({ submitted: true });
+
+        if (this.input && typeof this.props.value !== "string") {
+          this.input.value = "";
+        }
```

Comment:
Do we have use cases the `value` became non-string? It's clearly marked as `string` in props.

Reply:
With the second check I try to check if this is a controlled vs uncontrolled instance. Since I can only reset the value in the uncontrolled state.

Reply:
Thinking more about this maybe the method `getValue()` became misleading for now. As we see from it's name it should return a value, but shouldn't do any side effects like `this.input.value = "";`.

Reply:
And is this check relates to showing and icon for user? (I'm just wanted to reconsider do we really need this dropping value operation)?

Reply:
This isn't part of `getValue()` (though I admit that the github diff implies that it is). It is part of `onKeyDown`

Reply:
Oh yes, my fault. Good to go!